services/auth/test/gcp_test.js

```diff
@@ -1,88 +1,63 @@
+const _ = require('lodash');
 const assert = require('assert');
 const helper = require('./helper');
 const testing = require('taskcluster-lib-testing');
-const slugid = require('slugid');
-const {google} = require('googleapis');
 
-helper.secrets.mockSuite(testing.suiteName(), ['app', 'gcp'], function(mock, skipping) {
-  if (mock) {
-    return;
-  }
-
-  helper.withPulse('mock', skipping);
-  helper.withEntities('mock', skipping);
-  helper.withRoles('mock', skipping);
+helper.secrets.mockSuite(testing.suiteName(), ['app', 'gcp', 'azure'], function(mock, skipping) {
+  helper.withGcp(mock, skipping);
+  helper.withPulse(mock, skipping);
+  helper.withEntities(mock, skipping);
+  helper.withRoles(mock, skipping);
   helper.withServers(mock, skipping);
   helper.withCfg(mock, skipping);
 
-  let auth, account, iam;
-  const accountId = slugid.nice().replace(/_/g, '').toLowerCase();
-
-  suiteSetup('GCP credentials', async () => {
-    if (skipping()) {
-      return;
-    }
-    const credentials = helper.secrets.get('gcp').credentials;
-    const projectId = credentials.project_id;
-
-    auth = google.auth.fromJSON(credentials);
-
-    auth.scopes = [
-      'https://www.googleapis.com/auth/cloud-platform',
-      'https://www.googleapis.com/auth/iam',
-    ];
-
-    iam = google.iam('v1');
-
-    const res = await iam.projects.serviceAccounts.create({
-      auth,
-      name: `projects/${projectId}`,
-      resource: {
-        accountId,
-        serviceAccount: {
-          // This is a testing account and will be deleted by
-          // the end of the tests. If the test crashes, these
-          // accounts maybe left in IAM. Any account starting
-          // with taskcluster-auth-test- can be safely removed.
-          displayName: `taskcluster-auth-test-${accountId}`,
-        },
-      },
-    });
-
-    account = res.data;
-  });
-
-  suiteTeardown(async () => {
-    if (skipping()) {
-      return;
-    }
-
-    await iam.projects.serviceAccounts.delete({name: account.name, auth});
-  });
-
   test('gcpCredentials invalid account', async () => {
     try {
-      await helper.apiClient.gcpCredentials('-', 'invalidserviceaccount@mozilla.com');
-      assert.fail('The call should fail');
+      await helper.apiClient.gcpCredentials(helper.gcpAccount.project_id, 'invalid@mozilla.com');
     } catch (e) {
-      assert.equal(e.statusCode, 404);
+      if (e.statusCode !== 404) {
+        throw e;
+      }
```

Comment:
Why not a single line with `assert.equal`?

Reply:
In the case where the error is *not* a 404 error, `assert.equal` ends up hiding the error -- especially if the error doesn't have `statusCode`, as then the assertion just says `undefined !== 404`.  That's what prompted me to make this change: I messed up something and caused an error (I don't remember what) and was getting an assertion error that didn't tell me the issue.  Re-throwing that error directly showed me what I needed to fix.